v2/pkg/cli/delete.go

```diff
@@ -0,0 +1,288 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	_ "github.com/distribution/distribution/v3/registry/storage/driver/filesystem"
+	"github.com/google/uuid"
+
+	"github.com/openshift/oc-mirror/v2/pkg/api/v1alpha2"
+	"github.com/openshift/oc-mirror/v2/pkg/api/v1alpha3"
+	"github.com/openshift/oc-mirror/v2/pkg/archive"
+	"github.com/openshift/oc-mirror/v2/pkg/batch"
+	"github.com/openshift/oc-mirror/v2/pkg/config"
+	"github.com/openshift/oc-mirror/v2/pkg/delete"
+	clog "github.com/openshift/oc-mirror/v2/pkg/log"
+	"github.com/openshift/oc-mirror/v2/pkg/manifest"
+	"github.com/openshift/oc-mirror/v2/pkg/mirror"
+	"github.com/openshift/oc-mirror/v2/pkg/release"
+	"github.com/spf13/cobra"
+)
+
+const (
+	deleteErrMsg = "[delete] %v"
+)
+
+// NewDeleteCommand - setup all the relevant support structs
+// to eventually execute the 'delete' sub command
+func NewDeleteCommand(log clog.PluggableLoggerInterface) *cobra.Command {
```

Comment:
In v1 we are not duplicating all the code when a sub-command is needed, we add it as a sub-command in the same cmd ([link to the code here](https://github.com/openshift/oc-mirror/blob/03ce0ca797e73b6762fd3e24100ce043199519e9/pkg/cli/mirror/mirror.go#L140)):

cmd.AddCommand(version.NewVersionCommand(f, o.RootOptions))
cmd.AddCommand(list.NewListCommand(f, o.RootOptions))
cmd.AddCommand(describe.NewDescribeCommand(f, o.RootOptions))
cmd.AddCommand(initcmd.NewInitCommand(f, o.RootOptions))

Is it possible to follow the same approach here? 

@sherine-k the same we can change in the prepare command in a different PR.